igvm/commands.py

```diff
@@ -558,9 +598,16 @@ def _get_hypervisor(hostname, ignore_reserved=False):
             'Server "{0}" is online_reserved.'.format(dataset_obj['hostname'])
         )
 
-    return Hypervisor(dataset_obj)
+    hypervisor = Hypervisor(dataset_obj)
+    hypervisor.lock()
+
+    try:
+        yield hypervisor
+    finally:
+        hypervisor.unlock()
```

Comment:
Should we re-fetch in here as well?

Reply:
We could just to be safe, but we never change any HV attributes, unlike how it is done for VMs.